solr/core/src/java/org/apache/solr/cli/CLIUtils.java

```diff
@@ -226,6 +226,11 @@ public static String getZkHost(CommandLine cli) throws Exception {
       return zkHost;
     }
 
+//    zkHost = EnvUtils.getProperty("zkHost");
+//    if (zkHost != null && !zkHost.isBlank()) {
+//      return zkHost;
+//    }
```

Comment:
This is the fix, commented out to prove failing test. The `CLIUtils#getZkHost(cli)` util is used by many tools. Not sure if all tools document the support for `ZK_HOST`, but the `zk cp` tool does. Reading property `zkHost` works since `ZK_HOST` env is explicitly put into this sysProp by EnvUtils.